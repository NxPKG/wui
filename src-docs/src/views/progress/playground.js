import { PropTypes } from 'react-view';
import { WuiProgress } from '../../../../src/components/';
import { propUtilityForPlayground } from '../../services/playground';

export default () => {
  const docgenInfo = Array.isArray(WuiProgress.__docgenInfo)
    ? WuiProgress.__docgenInfo[0]
    : WuiProgress.__docgenInfo;
  const propsToUse = propUtilityForPlayground(docgenInfo.props);

  propsToUse.value = {
    ...propsToUse.value,
    value: undefined,
    type: PropTypes.Number,
  };

  propsToUse.valueText = {
    ...propsToUse.valueText,
    type: PropTypes.Boolean,
    value: false,
  };

  propsToUse.label = {
    ...propsToUse.label,
    type: PropTypes.String,
  };

  return {
    config: {
      componentName: 'WuiProgress',
      props: propsToUse,
      scope: {
        WuiProgress,
      },
      imports: {
        '@wazuh/wui': {
          named: ['WuiProgress'],
        },
      },
    },
  };
};
