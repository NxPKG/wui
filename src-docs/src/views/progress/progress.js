import React from 'react';

import { WuiProgress } from '../../../../src/components';

export default () => (
  <div>
    <WuiProgress size="xs" color="accent" />
  </div>
);
