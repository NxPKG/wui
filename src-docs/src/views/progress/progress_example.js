import React from 'react';

import { renderToHtml } from '../../services';

import { GuideSectionTypes } from '../../components';

import { WuiCallOut, WuiCode, WuiProgress } from '../../../../src/components';
import progressConfig from './playground';

import Progress from './progress';
const progressSource = require('!!raw-loader!./progress');
const progressHtml = renderToHtml(Progress);
const progressSnippet = '<WuiProgress size="xs" color="accent" />';

import ProgressValue from './progress_value';
const progressValueSource = require('!!raw-loader!./progress_value');
const progressValueHtml = renderToHtml(ProgressValue);
const progressValueSnippet = '<WuiProgress value={22} max={100} size="xs" />';

import ProgressFixed from './progress_fixed';
const progressFixedSource = require('!!raw-loader!./progress_fixed');
const progressFixedHtml = renderToHtml(ProgressFixed);
const progressFixedSnippet = `<!-- Position at top of parent container -->
<WuiProgress size="xs" color="accent" position="absolute" />

<!-- Position at top of screen, above global header -->
<WuiPortal>
  <WuiProgress size="xs" color="accent" position="fixed" />
</WuiPortal>`;

import ProgressSizeColor from './progress_size_color';
const progressSizeColorSource = require('!!raw-loader!./progress_size_color');
const progressSizeColorHtml = renderToHtml(ProgressSizeColor);
const progressSizeColorSnippet = `<WuiProgress
  value={20}
  max={100} 
  size="s"
  color="accent"
/>`;

import ProgressChart from './progress_chart';
const progressChartSource = require('!!raw-loader!./progress_chart');
const progressChartHtml = renderToHtml(ProgressChart);
const progressChartSnippet = `<WuiProgress 
  value={20}
  valueText={true}
  label={label}
  max={100} 
/>`;

export const ProgressExample = {
  title: 'Progress',
  sections: [
    {
      source: [
        {
          type: GuideSectionTypes.JS,
          code: progressSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: progressHtml,
        },
      ],
      text: (
        <p>
          The <strong>WuiProgress</strong> component by default will display in
          an indeterminate loading state (rendered as a single div) until you
          define a <WuiCode>max</WuiCode> and <WuiCode>value</WuiCode> prop. The{' '}
          <WuiCode>size</WuiCode> prop refers to its vertical height. It will
          always stretch <WuiCode>100%</WuiCode> to its container.
        </p>
      ),
      snippet: progressSnippet,
      props: { WuiProgress },
      demo: <Progress />,
    },
    {
      title: 'Progress with values',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: progressValueSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: progressValueHtml,
        },
      ],
      text: (
        <p>
          Once the <WuiCode>max</WuiCode> and <WuiCode>value</WuiCode> props are
          set, it will act as a determinate progress bar. This is rendered using
          an HTML5 <WuiCode>progress</WuiCode> tag.
        </p>
      ),
      snippet: progressValueSnippet,
      demo: <ProgressValue />,
    },
    {
      title: 'Progress can have absolute or fixed positions',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: progressFixedSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: progressFixedHtml,
        },
      ],
      text: (
        <div>
          <p>
            Using the <WuiCode>position</WuiCode> prop we can align our bar to
            be <WuiCode>fixed</WuiCode> or <WuiCode>absolute</WuiCode>. In both
            options, the background color of the base bar is dropped (since the
            context of width is already known from your wrapping element). For
            the absolute option, make sure that your wrapping element has{' '}
            <WuiCode language="sass">position: relative</WuiCode> applied.
          </p>
          <WuiCallOut
            title="Note about progress bars over fixed headers"
            iconType="iInCircle">
            <p>
              Using <strong>WuiProgress</strong> with a <WuiCode>fixed</WuiCode>{' '}
              position may result in it being overlayed when its parent wrapper
              has a <WuiCode>z-index</WuiCode> value lower than another fixed
              element, such as <strong>WuiHeader</strong>. In that case, wrap{' '}
              <strong>WuiProgress</strong> in an <strong>WuiPortal</strong> as
              seen on the Snippet tab.
            </p>
          </WuiCallOut>
        </div>
      ),
      snippet: progressFixedSnippet,
      demo: <ProgressFixed />,
    },
    {
      title: 'Progress has a range of sizes and colors',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: progressSizeColorSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: progressSizeColorHtml,
        },
      ],
      text: (
        <p>
          Both <WuiCode>size</WuiCode> and <WuiCode>color</WuiCode> can be
          provided as props. These values will work on both determinate and
          indeterminate progress bars.
        </p>
      ),
      demo: <ProgressSizeColor />,
      snippet: progressSizeColorSnippet,
    },
    {
      title: 'Progress for charts',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: progressChartSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: progressChartHtml,
        },
      ],
      text: (
        <div>
          <p>
            Determinate progress bar can be used as simple bar charts. Use them
            with the <WuiCode>label</WuiCode> and <WuiCode>valueText</WuiCode>{' '}
            props to show the data corresponding to each bar. The{' '}
            <WuiCode>valueText</WuiCode> renders as the same color as the{' '}
            <strong>WuiProgress</strong>.
          </p>
          <p>
            Setting <WuiCode language="ts">{'valueText={true}'}</WuiCode> will
            add a % sign next to the<WuiCode>value</WuiCode> passed. If you want
            to display a custom <WuiCode>valueText</WuiCode>, you can pass a
            node instead.
          </p>
        </div>
      ),
      demo: <ProgressChart />,
      snippet: progressChartSnippet,
    },
  ],
  playground: progressConfig,
};
