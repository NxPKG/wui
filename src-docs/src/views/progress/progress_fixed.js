import React, { useState, useEffect } from 'react';

import {
  WuiProgress,
  WuiSpacer,
  WuiButton,
  WuiText,
  WuiPanel,
  WuiCallOut,
  WuiFlexGroup,
  WuiFlexItem,
  WuiHeader,
  WuiHeaderLogo,
  WuiHeaderSection,
  WuiHeaderSectionItem,
  WuiPortal,
} from '../../../../src/components';

export default () => {
  const [value, setValue] = useState(0);
  const [showProgress, setShowProgress] = useState(false);
  const [showHeader, setShowHeader] = useState(false);

  let timer;
  const progress = value => {
    if (value > 100) {
      setValue(100);
    } else {
      setValue(value);
      const diff = Math.round(Math.random() * 10);
      timer = setTimeout(() => progress(value + diff), 250);
    }
  };

  const toggleProgress = () => {
    const currentState = showProgress;

    if (!currentState) {
      timer = setTimeout(() => progress(0), 250);
    } else {
      clearTimeout(timer);
      setValue(0);
    }
    setShowProgress(!showProgress);
    setShowHeader(false);
  };

  const toggleHeader = () => {
    setShowProgress(false);
    setShowHeader(!showHeader);
  };

  useEffect(() => {
    return () => {
      clearTimeout(timer);
    };
  }, [timer]);

  let progress2 = null;

  if (showProgress) {
    progress2 = (
      <div>
        <WuiCallOut title="Look up!" color="warning" iconType="sortUp">
          <p>The progress bar is fixed to the top of your browser.</p>
        </WuiCallOut>
        <WuiProgress value={value} max={100} size="s" position="fixed" />
      </div>
    );
  }

  if (showHeader) {
    progress2 = (
      <div>
        <WuiCallOut title="Look up!" color="warning" iconType="sortUp">
          <p>
            The progress bar is fixed to the top of your browser and positioned
            above an <strong>WuiHeader</strong>.
          </p>
        </WuiCallOut>
        <WuiHeader
          style={{ position: 'fixed', top: 0, left: 0, width: '100%' }}>
          <WuiHeaderSection grow={false}>
            <WuiHeaderSectionItem border="right">
              <WuiHeaderLogo
                iconType="logoWazuh"
                href="#"
                aria-label="Go to home page"
              />
            </WuiHeaderSectionItem>
          </WuiHeaderSection>
        </WuiHeader>
        <div style={{ position: 'absolute', zIndex: '5' }}>
          <WuiPortal>
            <WuiProgress size="xs" color="accent" position="fixed" />
          </WuiPortal>
        </div>
      </div>
    );
  }

  return (
    <div>
      <WuiPanel style={{ width: 300, position: 'relative' }}>
        <WuiProgress size="xs" color="accent" position="absolute" />
        <WuiText>
          <h2>Absolutely!</h2>
          <p>
            The progress bar is absolutely positioned in this panel. You could
            see how this might be useful in our Toast component.
          </p>
        </WuiText>
      </WuiPanel>

      <WuiSpacer size="l" />

      <WuiFlexGroup gutterSize="s" alignItems="center">
        <WuiFlexItem grow={false}>
          <WuiButton size="s" onClick={toggleProgress}>
            Toggle a fixed bar
          </WuiButton>
        </WuiFlexItem>
        <WuiFlexItem grow={false}>
          <WuiButton size="s" onClick={toggleHeader}>
            Toggle a fixed bar with header
          </WuiButton>
        </WuiFlexItem>
      </WuiFlexGroup>

      <WuiSpacer size="l" />

      {progress2}
    </div>
  );
};
