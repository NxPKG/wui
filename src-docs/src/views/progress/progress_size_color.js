import React from 'react';

import { WuiProgress, WuiSpacer } from '../../../../src/components';

export default () => (
  <div>
    <WuiProgress value={20} max={100} color="subdued" size="xs" />
    <WuiSpacer size="l" />

    <WuiProgress value={40} max={100} color="accent" size="xs" />
    <WuiSpacer size="l" />

    <WuiProgress value={60} max={100} color="primary" size="s" />
    <WuiSpacer size="l" />

    <WuiProgress value={80} max={100} color="secondary" size="m" />
    <WuiSpacer size="l" />

    <WuiProgress value={90} max={100} color="danger" size="l" />
  </div>
);
