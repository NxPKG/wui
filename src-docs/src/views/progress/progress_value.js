import React, { useState, useEffect } from 'react';

import {
  WuiButton,
  WuiFlexGroup,
  WuiFlexItem,
  WuiProgress,
  WuiText,
} from '../../../../src/components';

export default () => {
  const [value, setValue] = useState(0);
  const [showProgress, setShowProgress] = useState(false);

  let timer;
  const progress = value => {
    if (value > 100) {
      setValue(100);
    } else {
      setValue(value);
      const diff = Math.round(Math.random() * 10);
      timer = setTimeout(() => progress(value + diff), 250);
    }
  };
  const toggleProgress = () => {
    const currentState = showProgress;

    if (!currentState) {
      timer = setTimeout(() => progress(0), 250);
    } else {
      clearTimeout(timer);
      setValue(0);
    }
    setShowProgress(!showProgress);
  };

  useEffect(() => {
    return () => {
      clearTimeout(timer);
    };
  }, [timer]);

  return (
    <WuiFlexGroup alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButton size="s" onClick={toggleProgress}>
          Toggle progress
        </WuiButton>
      </WuiFlexItem>
      <WuiFlexItem grow={false}>
        <WuiText>
          <p>{value}</p>
        </WuiText>
      </WuiFlexItem>
      <WuiFlexItem>
        <WuiProgress value={value} max={100} size="xs" />
      </WuiFlexItem>
    </WuiFlexGroup>
  );
};
