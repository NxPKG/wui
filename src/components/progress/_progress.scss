// This file uses uncommon vendor prefixes not covered by our compilers
// sass-lint:disable no-vendor-prefixes

.wuiProgress {
  position: relative;
  overflow: hidden;
  background-color: $wuiColorLightShade;
}

// Progress bars come in different sizes.
$wuiProgressSizes: (
  xs: $wuiSizeXS / 2,
  s: $wuiSizeXS,
  m: $wuiSizeS,
  l: $wuiSize,
);

@each $name, $size in $wuiProgressSizes {
  .wuiProgress--#{$name} {
    height: $size;
  }
}

// https://css-tricks.com/html5-progress-element/
// Good resource if you need to work in here. There's some gotchas with
// dealing with cross-browser progress bars.
.wuiProgress--native {
  display: block;
  width: 100%;
  appearance: none;
  border: none;

  &::-webkit-progress-bar {
    background-color: $wuiColorLightShade;
  }

  &::-webkit-progress-value {
    transition: width $wuiAnimSpeedNormal linear;
  }

  &::-moz-progress-bar {
    transition: width $wuiAnimSpeedNormal linear;
  }
}

/**
 * An indeterminate bar has an unreliable end time. Because of a Firefox animation issue,
 * we apply this style to a <div> instead of a <progress> element.
 * See https://css-tricks.com/html5-progress-element/ for more info.
 */
.wuiProgress--indeterminate {
  &:before {
    position: absolute;
    content: '';
    width: 100%;
    top: 0;
    bottom: 0;
    left: 0;
    transform: scaleX(0) translateX(0%);
    animation: wuiProgress 1s $wuiAnimSlightResistance infinite;
  }
}

// Progress bars can be set to fixed or absolute.
.wuiProgress--fixed {
  position: fixed;
  z-index: $wuiZHeader + 1;
}

.wuiProgress--absolute {
  position: absolute;
}

.wuiProgress--fixed,
.wuiProgress--absolute {
  top: 0;
  left: 0;
  right: 0;
  background-color: transparent;

  &.wuiProgress--native {
    &::-webkit-progress-bar {
      background-color: transparent;
    }

    &::-moz-progress-bar {
      background-color: transparent;
    }
  }
}

// Progress bars come in different colors.
$wuiProgressColors: (
  primary: $wuiColorPrimary,
  secondary: $wuiColorSecondary,
  warning: $wuiColorWarning,
  danger: $wuiColorDanger,
  accent: $wuiColorAccent,
  subdued: $wuiTextSubduedColor,
);

@each $name, $color in $wuiProgressColors {
  .wuiProgress--#{$name} {
    &.wuiProgress--native {
      &::-webkit-progress-value {
        background-color: $color;
      }

      &::-moz-progress-bar {
        background-color: $color;
      }
    }

    &.wuiProgress--indeterminate {
      &:before {
        background-color: $color;
      }
    }
  }

  .wuiProgress__data--#{$name} {
    .wuiProgress__valueText {
      color: makeHighContrastColor($color);
    }
  }
}

@keyframes wuiProgress {
  0% {
    transform: scaleX(1) translateX(-100%);
  }

  100% {
    transform: scaleX(1) translateX(100%);
  }
}

.wuiProgress__data {
  display: flex;
  justify-content: space-between;
}

.wuiProgress__label,
.wuiProgress__valueText {
  @include wuiText;
  @include wuiFontSizeXS;
  @include wuiTextTruncate;
}

.wuiProgress__label {
  flex-grow: 1;

  // Only restrict the valueText if it's the sibling of the label
  // Gives width precedence to the value text forcing consumers to round their values
  + .wuiProgress__valueText {
    padding-left: $wuiSizeXS;
    flex-grow: 1;
    text-align: right;
    flex-shrink: 0;
  }
}

.wuiProgress__valueText {
  // Tabular numbers ensure the line up nicely when right-aligned
  font-feature-settings: 'tnum' 1;
  margin-left: auto;
}

.wuiProgress__data--l {
  .wuiProgress__label,
  .wuiProgress__valueText {
    @include wuiFontSizeS;
  }
}
