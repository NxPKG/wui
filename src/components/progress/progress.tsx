/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, {
  FunctionComponent,
  HTMLAttributes,
  ProgressHTMLAttributes,
  ReactNode,
  Fragment,
} from 'react';
import classNames from 'classnames';
import { WuiI18n } from '../i18n';
import { WuiInnerText } from '../inner_text';
import { CommonProps, ExclusiveUnion } from '../common';
import { isNil } from '../../services/predicate';

const sizeToClassNameMap = {
  xs: 'wuiProgress--xs',
  s: 'wuiProgress--s',
  m: 'wuiProgress--m',
  l: 'wuiProgress--l',
};

export const SIZES = Object.keys(sizeToClassNameMap);

export type WuiProgressSize = keyof typeof sizeToClassNameMap;

const colorToClassNameMap = {
  primary: 'wuiProgress--primary',
  secondary: 'wuiProgress--secondary',
  danger: 'wuiProgress--danger',
  subdued: 'wuiProgress--subdued',
  accent: 'wuiProgress--accent',
};

export const COLORS = Object.keys(colorToClassNameMap);

export type WuiProgressColor = keyof typeof colorToClassNameMap;

const dataColorToClassNameMap = {
  primary: 'wuiProgress__data--primary',
  secondary: 'wuiProgress__data--secondary',
  danger: 'wuiProgress__data--danger',
  subdued: 'wuiProgress__data--subdued',
  accent: 'wuiProgress__data--accent',
};

const positionsToClassNameMap = {
  fixed: 'wuiProgress--fixed',
  absolute: 'wuiProgress--absolute',
  static: '',
};

export const POSITIONS = Object.keys(positionsToClassNameMap);

export type WuiProgressPosition = keyof typeof positionsToClassNameMap;

export type WuiProgressProps = CommonProps & {
  size?: WuiProgressSize;
  color?: WuiProgressColor;
  position?: WuiProgressPosition;
};

type Indeterminate = WuiProgressProps & HTMLAttributes<HTMLDivElement>;

type Determinate = WuiProgressProps &
  Omit<ProgressHTMLAttributes<HTMLProgressElement>, 'max'> & {
    max?: number;
    /*
     * If true, will render the percentage, otherwise pass a custom node
     */
    valueText?: boolean | ReactNode;
    label?: ReactNode;
    /**
     * Object of props passed to the <span/> wrapping the determinate progress's label
     */
    labelProps?: HTMLAttributes<HTMLSpanElement>;
  };

export const WuiProgress: FunctionComponent<ExclusiveUnion<
  Determinate,
  Indeterminate
>> = ({
  className,
  color = 'secondary',
  size = 'm',
  position = 'static',
  max,
  valueText = false,
  label,
  value,
  labelProps,
  ...rest
}) => {
  const determinate = !isNil(max);
  const classes = classNames(
    'wuiProgress',
    {
      'wuiProgress--indeterminate': !determinate,
      'wuiProgress--native': determinate,
    },
    sizeToClassNameMap[size],
    colorToClassNameMap[color],
    positionsToClassNameMap[position],
    className
  );
  const dataClasses = classNames(
    'wuiProgress__data',
    {
      'wuiProgress__data--l': size === 'l',
    },
    dataColorToClassNameMap[color]
  );
  const labelClasses = classNames(
    'wuiProgress__label',
    labelProps && labelProps.className
  );

  let valueRender: ReactNode;
  if (valueText === true) {
    // valueText is true
    valueRender = (
      <WuiI18n
        token="wuiProgress.valueText"
        default="{value}%"
        values={{
          value,
        }}
      />
    );
  } else if (valueText) {
    // valueText exists
    valueRender = valueText;
  }

  // Because of a Firefox animation issue, indeterminate progress needs to not use <progress />.
  // See https://css-tricks.com/html5-progress-element/

  if (determinate) {
    return (
      <Fragment>
        {label || valueText ? (
          <div className={dataClasses}>
            {label && (
              <WuiInnerText>
                {(ref, innerText) => (
                  <span
                    title={innerText}
                    ref={ref}
                    {...labelProps}
                    className={labelClasses}>
                    {label}
                  </span>
                )}
              </WuiInnerText>
            )}
            {valueRender && (
              <WuiInnerText>
                {(ref, innerText) => (
                  <span
                    title={innerText}
                    ref={ref}
                    className="wuiProgress__valueText">
                    {valueRender}
                  </span>
                )}
              </WuiInnerText>
            )}
          </div>
        ) : (
          undefined
        )}
        <progress
          className={classes}
          max={max}
          value={value}
          aria-hidden={label && valueText ? true : false}
          {...(rest as ProgressHTMLAttributes<HTMLProgressElement>)}
        />
      </Fragment>
    );
  } else {
    return (
      <div className={classes} {...(rest as HTMLAttributes<HTMLDivElement>)} />
    );
  }
};
