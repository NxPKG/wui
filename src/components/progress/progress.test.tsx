/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { WuiProgress } from './progress';

describe('WuiProgress', () => {
  test('is rendered', () => {
    const component = render(<WuiProgress {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  test('has max', () => {
    const component = render(<WuiProgress max={100} {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  test('has value', () => {
    const component = render(<WuiProgress value={100} {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  test('is determinate', () => {
    const val = 50;
    const component = render(
      <WuiProgress max={val ? 100 : undefined} value={val} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('is indeterminate', () => {
    const val = undefined;
    const component = render(
      <WuiProgress max={val ? 100 : undefined} value={val} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('has valueText and label', () => {
    const component = render(
      <WuiProgress
        valueText="150"
        label="Label"
        value={50}
        max={100}
        {...requiredProps}
      />
    );

    expect(component).toMatchSnapshot();
  });

  test('valueText is true', () => {
    const component = render(
      <WuiProgress valueText={true} value={50} max={100} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('has labelProps', () => {
    const component = render(
      <WuiProgress
        max={100}
        value={50}
        labelProps={{ title: 'Custom title' }}
        valueText="150"
        {...requiredProps}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
