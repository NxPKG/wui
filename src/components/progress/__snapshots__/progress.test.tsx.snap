// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WuiProgress has labelProps 1`] = `
Array [
  <div
    class="wuiProgress__data wuiProgress__data--secondary"
  >
    <span
      class="wuiProgress__valueText"
    >
      150
    </span>
  </div>,
  <progress
    aria-hidden="false"
    aria-label="aria-label"
    class="wuiProgress wuiProgress--native wuiProgress--m wuiProgress--secondary testClass1 testClass2"
    data-test-subj="test subject string"
    max="100"
    value="50"
  />,
]
`;

exports[`WuiProgress has max 1`] = `
<progress
  aria-hidden="false"
  aria-label="aria-label"
  class="wuiProgress wuiProgress--native wuiProgress--m wuiProgress--secondary testClass1 testClass2"
  data-test-subj="test subject string"
  max="100"
/>
`;

exports[`WuiProgress has value 1`] = `
<div
  aria-label="aria-label"
  class="wuiProgress wuiProgress--indeterminate wuiProgress--m wuiProgress--secondary testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`WuiProgress has valueText and label 1`] = `
Array [
  <div
    class="wuiProgress__data wuiProgress__data--secondary"
  >
    <span
      class="wuiProgress__label"
    >
      Label
    </span>
    <span
      class="wuiProgress__valueText"
    >
      150
    </span>
  </div>,
  <progress
    aria-hidden="true"
    aria-label="aria-label"
    class="wuiProgress wuiProgress--native wuiProgress--m wuiProgress--secondary testClass1 testClass2"
    data-test-subj="test subject string"
    max="100"
    value="50"
  />,
]
`;

exports[`WuiProgress is determinate 1`] = `
<progress
  aria-hidden="false"
  aria-label="aria-label"
  class="wuiProgress wuiProgress--native wuiProgress--m wuiProgress--secondary testClass1 testClass2"
  data-test-subj="test subject string"
  max="100"
  value="50"
/>
`;

exports[`WuiProgress is indeterminate 1`] = `
<div
  aria-label="aria-label"
  class="wuiProgress wuiProgress--indeterminate wuiProgress--m wuiProgress--secondary testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`WuiProgress is rendered 1`] = `
<div
  aria-label="aria-label"
  class="wuiProgress wuiProgress--indeterminate wuiProgress--m wuiProgress--secondary testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`WuiProgress valueText is true 1`] = `
Array [
  <div
    class="wuiProgress__data wuiProgress__data--secondary"
  >
    <span
      class="wuiProgress__valueText"
    >
      50%
    </span>
  </div>,
  <progress
    aria-hidden="false"
    aria-label="aria-label"
    class="wuiProgress wuiProgress--native wuiProgress--m wuiProgress--secondary testClass1 testClass2"
    data-test-subj="test subject string"
    max="100"
    value="50"
  />,
]
`;
